src/EditorFeatures/Core.Wpf/StringIndentation/StringIndentationTaggerProvider.cs

```diff
@@ -25,120 +23,101 @@
-            IAsynchronousOperationListenerProvider listenerProvider)
-            : base(threadingContext, globalOptions, visibilityTracker, listenerProvider.GetListener(FeatureAttribute.StringIndentation))
-        {
-            _editorFormatMap = editorFormatMapService.GetEditorFormatMap("text");
-        }
+        // Note: we don't listen for OnTextChanged.  They'll get reported by the ViewSpan changing. 
+        return TaggerEventSources.Compose(
+            TaggerEventSources.OnViewSpanChanged(ThreadingContext, textView),
+            new EditorFormatMapChangedEventSource(_editorFormatMap));
+    }
 
-        protected override TaggerDelay EventChangeDelay => TaggerDelay.NearImmediate;
-
-        /// <summary>
-        /// We want the span tracking mode to be inclusive here.  That way if the user types space here:
-        /// 
-        /// <code>
-        /// var v = """
-        ///            goo
-        ///         """
-        ///        ^ // here
-        /// </code>
-        /// 
-        /// then the span of the tag will grow to the right and the line will immediately redraw in the correct position
-        /// while we're in the process of recomputing the up to date tags.
-        /// </summary>
-        protected override SpanTrackingMode SpanTrackingMode => SpanTrackingMode.EdgeInclusive;
-
-        protected override ITaggerEventSource CreateEventSource(
-            ITextView textView, ITextBuffer subjectBuffer)
-        {
-            // Note: we don't listen for OnTextChanged.  They'll get reported by the ViewSpan changing. 
-            return TaggerEventSources.Compose(
-                TaggerEventSources.OnViewSpanChanged(ThreadingContext, textView),
-                new EditorFormatMapChangedEventSource(_editorFormatMap));
-        }
+    protected override async Task ProduceTagsAsync(
+        TaggerContext<StringIndentationTag> context, DocumentSnapshotSpan documentSnapshotSpan, CancellationToken cancellationToken)
+    {
+        var document = documentSnapshotSpan.Document;
+        if (document == null)
+            return;
 
-        protected override IEnumerable<SnapshotSpan> GetSpansToTag(ITextView? textView, ITextBuffer subjectBuffer)
-        {
-            this.ThreadingContext.ThrowIfNotOnUIThread();
-            Contract.ThrowIfNull(textView);
-
-            // Find the visible span some 100 lines +/- what's actually in view.  This way
-            // if the user scrolls up/down, we'll already have the results.
-            var visibleSpanOpt = textView.GetVisibleLinesSpan(subjectBuffer, extraLines: 100);
-            if (visibleSpanOpt == null)
-            {
-                // Couldn't find anything visible, just fall back to tagging all brace locations
-                return base.GetSpansToTag(textView, subjectBuffer);
-            }
-
-            return SpecializedCollections.SingletonEnumerable(visibleSpanOpt.Value);
-        }
```

Comment:
computing hte spans to tag is no longer the responsibility of 'AsynchronousViewportTaggerProvider's.  The base class handles this.